src/analysisd/decoders/syscollector.c

```diff
@@ -1457,6 +1469,12 @@ int decode_package( Eventinfo *lf,cJSON * logJSON,int *socket) {
             wm_strcat(&msg, "NULL", '|');
         }
 
+        //Calculating hash of the NULL terminated array
+        fields_to_hash[3] = NULL;
+        wdbi_sha_calculation((const char **)fields_to_hash, hexdigest, 0);
```

Comment:
Can't we solve this with a similar approach to syscollector.c?
```
wdbi_sha_calculation(NULL, hexdigest,3,
                         architecture->valuestring ? architecture->valuestring : "",
                         name->valuestring ? name->valuestring : "",
                         version->valuestring ? version->valuestring : "");
```
                        
It will simplify the memory allocation and free.

Reply:
+1

Reply:
Now, the two calls are made using the same method. And the **count** parameter was suppressed.